tensorflow_datasets/core/features/audio_feature_test.py

```diff
@@ -66,6 +66,21 @@ def test_numpy_array_float(self):
         ],
     )
 
+  def test_to_json(self):
```

Comment:
Would it be possible to some-how automate this test inside `assertFeature` ?

Rather than checking individual fields, I think we could check that exporting/loading the feature yield the same results:

```py
def assertFeature(feature, *kwargs):
  with subtest('feature'):
    _assert_feature(feature, *kwargs)
  with subtest('exported_imported'):
    new_feature = FeatureConnector.from_json(feature.to_json())
    _assert_feature(new_feature, *kwargs)
```

This way all features would be tested automatically. And we don't need to test the intermediate representation. If the reconstructed feature works, it's good !

Reply:
Added a similar test.
```
  def _assert_feature(self, feature):
    new_feature = feature.from_json_content(feature.to_json_content())
    self.assertEqual(str(feature), str(new_feature))
```

Reply:
> Added a similar test.

This test is good, but feel fragile and don't really test that the feature is correctly exported/imported.

For instance, with ClassLabel, your implementation would not test that label names are correctly restored.